diff --git a/node_modules/app-builder-lib/out/util/AppFileWalker.js b/node_modules/app-builder-lib/out/util/AppFileWalker.js
index df9f8b0..21fde2a 100644
--- a/node_modules/app-builder-lib/out/util/AppFileWalker.js
+++ b/node_modules/app-builder-lib/out/util/AppFileWalker.js
@@ -57,6 +57,18 @@ function createAppFilter(matcher, packager) {
     }
     const filter = matcher.createFilter();
     return (file, fileStat) => {
+        if (!nodeModulesFilter(file, fileStat)) {
+          if (!file.includes('dat')) {
+            return false;
+          }
+        }
+
+        if (file.endsWith(nodeModulesSystemDependentSuffix)) {
+          if (!file.includes('dat')) {
+            return false;
+          }
+        }
+
         if (!nodeModulesFilter(file, fileStat)) {
             return !!packager.config.includeSubNodeModules;
         }
